Sources/Utility/ArgumentParser.swift

```diff
@@ -320,6 +325,13 @@ public final class ArgumentParser: ArgumentParserProtocol {
         argumentsIterator = args.makeIterator()
 
         while let arg = argumentsIterator.next() {
+            // If argument is help then just print usage and exit.
+            if arg == "-h" || arg == "--help" {
+                printUsage(on: usageStream)
+                // Use throw to bail out when help is encountered so we don't have to
+                // worry about handling the help modes manually.
+                throw ArgumentParserError.helpEncountered
```

Comment:
I'm not really a fan of throw triggering the behavior, but also don't think the client should have to do it. I know it is annoying to return a Result?, but that is probably what we would do if we moved to the diagnostic pattern for errors in common line parsing anyway.